File: LearnSparkJvm/src/main/java/com/dt/learn/word/WordSparkApplication.java
package com.dt.learn.word;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.storage.StorageLevel;
import scala.Tuple2;

import java.util.Arrays;

/**
 * @author dt 2018/7/11 12:26
 */
public class WordSparkApplication {

    public static void main(String[] args) {

        SparkConf conf = new SparkConf().setMaster("local").setAppName("wordCount");
        JavaSparkContext sc = new JavaSparkContext(conf);
        JavaRDD<String> rdd = sc.textFile("/tmp/sparkStudy/thatGirl.txt");
        JavaRDD<String> stringJavaRDD = rdd.flatMap(x -> Arrays.asList(x.split(" ")).iterator());

        JavaPairRDD<String, Integer> wordCount = stringJavaRDD.mapToPair(x -> {
            return new Tuple2<>(x, 1);
        }).reduceByKey((x, y) -> {
            return x + y;
        });

        /*将数据直接保存到缓存中*/
        wordCount.persist(StorageLevel.MEMORY_AND_DISK());

        wordCount.saveAsTextFile("/tmp/sparkStudy/javaword");

        /*直接清除缓存*/
        wordCount.unpersist();

    }

}

File: LearnSparkJvm/src/main/scala/com/dt/learn/word/WordSparkApplication.scala
package com.dt.learn.word

import org.apache.spark.{SparkConf, SparkContext}

/**
  * @author dt 2018/7/15 18:28
  */
object WordSparkApplication {

  def main(args: Array[String]): Unit = {

    val conf = new SparkConf().setAppName("wordCount")
    val sc = new SparkContext(conf)
    val input = sc.textFile("/tmp/sparkStudy/thatGirl.txt")
    val words = input.flatMap(x => x.split(" "))
    val counts = words.map(word => (word,1)).reduceByKey((v1,v2) => v1+v2)
    /*保存到缓存中*/
    counts.persist()
    counts.saveAsTextFile("/tmp/sparkStudy/word")
    /*清楚缓存中的数据*/
    counts.unpersist()
    sc.stop()

  }

}
